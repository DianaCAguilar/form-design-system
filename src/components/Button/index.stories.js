import React from 'react';
import State from 'util/storybook-docs/State';
import { StoryWrapper, StoryItem } from 'util/storybook-docs/StoryLayout';
import StarFilledIcon from 'lib/icons/react/StarFilledIcon';
import WorldIcon from 'lib/icons/react/WorldIcon';
import Button, { THEMES, SIZES } from '.';

const Template = (args) => <Button label="Button" {...args} />;

export const Primary = Template.bind({});

export const Themes = () => (
  <StoryWrapper>
    {THEMES.map((theme, idx) => (
      <StoryItem key={idx}>
        <Button theme={theme} label={theme} />
      </StoryItem>
    ))}
  </StoryWrapper>
);

export const isDestructive = () => (
  <StoryWrapper>
    {THEMES.map((theme, idx) => (
      <StoryItem key={idx}>
        <Button isDestructive theme={theme} label={theme} />
      </StoryItem>
    ))}
  </StoryWrapper>
);

isDestructive.parameters = {
  docs: {
    description: {
      story:
        'Modifies the theme. Should only be used when the interaction involves data loss, and should also prompt the user to confirm the action.',
    },
  },
};

export const Inverted = () => (
  <div className="inverted bgColor--blue padding--all rounded--all">
    <StoryWrapper>
      <StoryItem>
        <Button theme="ghost" label="Button" />
      </StoryItem>
      <StoryItem>
        <Button theme="outlined" label="Button" />
      </StoryItem>
    </StoryWrapper>
  </div>
);

Inverted.parameters = {
  docs: {
    description: {
      story:
        'We currently support inverting `ghost` and `outlined` theme buttons. Simply surround the button(s) with an `.inverted` class.',
    },
  },
};

export const isActive = () => (
  <>
    <StoryWrapper>
      <State
        initialValue={{}}
        render={(value, setValue) => (
          <>
            {THEMES.map((theme, idx) => (
              <StoryItem key={idx}>
                <Button
                  theme={theme}
                  isActive={!value[idx]}
                  label={theme}
                  onClick={() => setValue({ ...value, [idx]: !value[idx] })}
                />
              </StoryItem>
            ))}
          </>
        )}
      />
      <State
        initialValue={{}}
        render={(value, setValue) => (
          <>
            {THEMES.map((theme, idx) => (
              <StoryItem key={idx}>
                <Button
                  theme={theme}
                  isActive={!value[idx]}
                  isDestructive
                  label={theme}
                  onClick={() => setValue({ ...value, [idx]: !value[idx] })}
                />
              </StoryItem>
            ))}
          </>
        )}
      />
    </StoryWrapper>
    <div className="inverted bgColor--blue padding--all rounded--all margin--top">
      <StoryWrapper>
        <State
          initialValue={{}}
          render={(value, setValue) => (
            <>
              {['ghost', 'outlined'].map((theme, idx) => (
                <StoryItem key={idx}>
                  <Button
                    theme={theme}
                    isActive={!value[idx]}
                    label={theme}
                    onClick={() => setValue({ ...value, [idx]: !value[idx] })}
                  />
                </StoryItem>
              ))}
            </>
          )}
        />
      </StoryWrapper>
    </div>
  </>
);

isActive.parameters = {
  docs: {
    description: {
      story:
        '`isActive` should be used on buttons to signify the button having some affect (applying a filter, causing a popover to be open). **Click to toggle on and off**.',
    },
  },
};

export const Sizes = () =>
  SIZES.map((size, idx) => (
    <div
      key={idx}
      className="display--inlineBlock margin--right--half margin--bottom--half"
    >
      <Button size={size} label="Button" />
    </div>
  ));

export const FullWidth = Template.bind({});
FullWidth.args = {
  isFullWidth: true,
};

export const Icons = () => (
  <StoryWrapper>
    <StoryItem>
      <Button IconLeft={WorldIcon} label="Button" />
    </StoryItem>
    <StoryItem>
      <Button IconRight={StarFilledIcon} label="Button" />
    </StoryItem>
  </StoryWrapper>
);

Icons.parameters = {
  docs: {
    description: {
      story:
        'ONLY pass in FDS Icon references to `IconLeft` or `IconRight`. Very important. (refer to code sample)',
    },
  },
};

export const Caret = () => (
  <StoryWrapper>
    <StoryItem>
      <Button theme="blue" hasCaret label="Caret" />
    </StoryItem>
    <StoryItem>
      <Button theme="blue" hasCaret IconRight={StarFilledIcon} label="Caret" />
    </StoryItem>
    <StoryItem>
      <Button theme="blue" hasCaret IconRight={StarFilledIcon} label="Caret" />
    </StoryItem>
  </StoryWrapper>
);

export const Loading = () => (
  <>
    <StoryWrapper>
      {THEMES.map((theme, idx) => (
        <StoryItem key={idx}>
          <Button theme={theme} isLoading label="Button" />
        </StoryItem>
      ))}
      {THEMES.map((theme, idx) => (
        <StoryItem key={idx}>
          <Button theme={theme} isDestructive isLoading label="Button" />
        </StoryItem>
      ))}
    </StoryWrapper>
    <div className="inverted bgColor--blue padding--all rounded--all margin--top">
      <StoryWrapper>
        <StoryItem>
          <Button theme="ghost" isLoading label="Button" />
        </StoryItem>
        <StoryItem>
          <Button theme="outlined" isLoading label="Button" />
        </StoryItem>
      </StoryWrapper>
    </div>
  </>
);

Loading.parameters = {
  docs: {
    description: {
      story:
        'Loading states should be shown whenever an action has the ability to take more than a couple of seconds of loading time',
    },
  },
};

export const Breakout = (args) => (
  <StoryWrapper>
    <StoryItem>
      <div
        className="alignChild--center--center padding--all--xl"
        style={{ border: '1px dashed #f09', position: 'relative' }}
      >
        <Button {...args} label="Button" isBreakoutLink />
      </div>
    </StoryItem>
  </StoryWrapper>
);

Breakout.parameters = {
  docs: {
    description: {
      story: 'Breakout buttons extend their clickable area to the next nearest parent.',
    },
  },
};

export const Misc = () => (
  <div style={{ width: '100px' }}>
    <Button label="Text can wrap" />
  </div>
);

export default {
  component: Button,
  title: 'components/Button',
  parameters: {
    docs: {
      attributes: ['rest'],
    },
  },
};
