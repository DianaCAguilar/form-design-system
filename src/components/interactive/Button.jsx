import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import baseElement from '../../util/baseElement';

const Button = ({ theme, isLoading, iconPlacement, Icon, Link, children, ...rest }) => {
  const Element = baseElement({ href: rest.href, as: Link });
  return (
    <Element
      className={cx('fdsButton', {
        'fdsButton--contained-blue': theme === 'contained--blue',
        'fdsButton--contained-red': theme === 'contained--red',
        'fdsButton--loading': isLoading,
      })}
      {...rest}
    >
      {Icon && iconPlacement === 'left' && (
        <div className="margin--right--half fdsButton-icon">
          <Icon size="xs" />
        </div>
      )}
      <span className={cx({ 'fdsButton--hiddenLabel': isLoading })}>{children}</span>
      {Icon && iconPlacement === 'right' && (
        <div className="margin--left--half fdsButton-icon">
          <Icon size="xs" />
        </div>
      )}
    </Element>
  );
};

Button.defaultProps = {
  theme: 'primary--blue',
  iconPlacement: 'right',
};

Button.propTypes = {
  /**
   * Pass **only** react-router `Link` here. You may **not**
   * pass anything else here: SFC, Class Component, etc (even
   * if they use react-router `Link` underneath the hood).
   */
  Link: PropTypes.func,
  /**
   * Controls whether loading spinner displays. Button text and icons are set to visibility hidden
   * to preserve the space set, whilst hiding them
   */
  isLoading: PropTypes.bool,
  /** Controls which side the `Icon` renders on, assuming you pass it */
  iconPlacement: PropTypes.oneOf(['left', 'right']),
  /** Used to render a FDS Icon (should only be used for FDS Icons) */
  Icon: PropTypes.func,
  /** Used to control the display and theme of the button */
  theme: PropTypes.oneOf(['container--blue', 'container--red']),
  /** Contents inside the button */
  children: PropTypes.node,
};

export default Button;
