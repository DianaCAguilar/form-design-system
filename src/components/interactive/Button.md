# Button

Displays a standard Button.
