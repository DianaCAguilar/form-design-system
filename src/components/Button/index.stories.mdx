
import withFdsIconWrapper from 'icons/templates/withFdsIconWrapper';
import { Meta, Preview, Story, Props } from "@storybook/addon-docs/blocks";
import {
  withKnobs,
  text,
  boolean,
  optionsKnob as options
} from "@storybook/addon-knobs";
import cool from "assets/ay.svg"

import State from 'util/storybook-docs/State';
import ImportPath from 'util/storybook-docs/ImportPath';
import { StoryWrapper, StoryItem } from 'util/storybook-docs/StoryLayout';
import { arrayToOptions } from 'util/storybook-docs/knobHelpers';
import Button, { THEMES } from "./";
import IconButton from "components/IconButton";
import StarFilledIcon from "lib/icons/react/StarFilledIcon";
import ButtonGroup from "components/ButtonGroup";

export const SIZES = ['s', 'm'];

<Meta title="Components/Button" decorators={[withKnobs]} component={Button} />

# Button

<ImportPath component={Button} />

<Story name="Knobs" parameters={{ docs: { disable: true }}}>
  <div className={boolean("Inverted", false) ? 'inverted' : ''}>
    <Button
      disabled={boolean("disabled", undefined)}
      isLoading={boolean("isLoading", undefined)}
      isDestructive={boolean("isDestructive", undefined)}
      isActive={boolean("isActive", undefined)}
      hasCaret={boolean("hasCaret", undefined)}
      IconLeft={boolean("IconLeft") ? StarFilledIcon : undefined}
      IconRight={boolean("IconRight") ? StarFilledIcon : undefined}
      isFullWidth={boolean("isFullWidth", undefined)}
      size={options("size", arrayToOptions(SIZES), undefined, {
        display: "inline-radio"
      })}
      theme={options(
        "theme",
        arrayToOptions(THEMES),
        undefined,
        {
          display: "inline-radio"
        }
      )}
      label={text("text", "Button")}
    />
  </div>
</Story>

<Preview isExpanded>
  <Button label="Button" />
</Preview>

<Props of={Button} />

### Themes

<Preview>
  <Story name="Themes">
    <StoryWrapper>
    <IconButton Icon={StarFilledIcon} label="Icon" />
    {THEMES.map((theme, idx) => (
      <StoryItem key={idx}>
        <Button
          theme={theme}
          label={theme}
        />
      </StoryItem>
    ))}
    </StoryWrapper>
  </Story>
</Preview>


### isDestructive

Modifies the theme. Should only be used when the interaction involves data loss, and should also prompt the user to confirm the action.

<Preview>
  <Story name="Destructive">
    <StoryWrapper>
    {THEMES.map(theme => (
      <StoryItem>
        <Button
          theme={theme}
          isDestructive
          label={theme}
        />
      </StoryItem>
    ))}
    </StoryWrapper>
  </Story>
</Preview>

 ### Inverted

We currently support inverting `ghost` and `outlined` theme
buttons. Simply surround the button(s) with an `.inverted` class.

<Preview>
  <Story name="Inverted">
    <div class="inverted bgColor--blue padding--all rounded--all">
      <StoryWrapper>
        <StoryItem>
          <Button theme="ghost" label="Button" />
        </StoryItem>
        <StoryItem>
          <Button theme="outlined" label="Button" />
        </StoryItem>
      </StoryWrapper>
    </div>
  </Story>
</Preview> 


### isActive

`isActive` should be used on buttons to signify the button having
some affect (applying a filter, causing a popover to be open).
**Click to toggle on and off**.

<Preview>
  <Story name="isActive">
    <>
      <StoryWrapper>
        <State
          initialValue={{}}
          render={(value, setValue) => ( 
            <>
              {THEMES.map((theme, idx) => (
                <StoryItem>
                  <Button
                    theme={theme}
                    isActive={!value[idx]}
                    label={theme}
                    onClick={() => setValue({ ...value, [idx]: !Boolean(value[idx])})}
                  />
                </StoryItem>
              ))}
            </>
          )}
        />
        <State
          initialValue={{}}
          render={(value, setValue) => ( 
            <>
              {THEMES.map((theme, idx) => (
                <StoryItem>
                  <Button
                    theme={theme}
                    isActive={!value[idx]}
                    isDestructive
                    label={theme}
                    onClick={() => setValue({ ...value, [idx]: !Boolean(value[idx])})}
                  />
                </StoryItem>
              ))}
            </>
          )}
        />
      </StoryWrapper>
      <div class="inverted bgColor--blue padding--all rounded--all margin--top">
        <StoryWrapper>
          <State
            initialValue={{}}
            render={(value, setValue) => ( 
              <>
                {['ghost', 'outlined'].map((theme, idx) => (
                  <StoryItem>
                    <Button
                      theme={theme}
                      isActive={!value[idx]}
                      label={theme}
                      onClick={() => setValue({ ...value, [idx]: !Boolean(value[idx])})}
                    />
                  </StoryItem>
                ))}
              </>
            )}
          />
        </StoryWrapper>
      </div>
    </>
  </Story>
</Preview>

### Sizes

<Preview>
  <Story name="Button Sizes">
    {SIZES.map(size => (
      <div className="display--inlineBlock margin--right--half margin--bottom--half">
        <Button
          size={size}
          label="Button"
        />
      </div>
    ))}
  </Story>
</Preview>

### Full width button

<Preview>
  <Story name="Full Width">
    <Button
      isFullWidth
      label="Button"
    />
  </Story>
</Preview>

### Icons

ONLY pass in FDS Icon references to `IconLeft` or `IconRight`. Very important. (refer to code sample)

<Preview>
  <Story name="Icons">
    <StoryWrapper>
      <StoryItem>
        <Button IconLeft={StarFilledIcon} label="Button" />
      </StoryItem>
      <StoryItem>
        <Button IconRight={StarFilledIcon} label="Button" />
      </StoryItem>
      <StoryItem>
        <IconButton Icon={StarFilledIcon} label="Button" />
      </StoryItem>
    </StoryWrapper>
  </Story>
</Preview>

### Caret

export const AY = props => (
<svg width="18px" height="20px" viewBox="0 0 18 20" version="1.1" xmlns="http://www.w3.org/2000/svg">
    <title>FTP / 01 Light / 01 Primary </title>
    <g id="Saved-Search" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd" fill-opacity="0.87">
        <g id="SFTP---01" transform="translate(-188.000000, -419.000000)" fill="#404040">
            <g id="Shape" transform="translate(185.000000, 417.000000)">
                <path d="M16.0002136,2 L21,6.99978644 L21,20 C21,21.0531354 20.1829516,21.9180536 19.1491027,21.9945067 L19,22 L7,22 C5.94686456,22 5.08194636,21.1829516 5.00549334,20.1491027 L5,20 L5,19 L7,19 L7,20 L19,20 L19,8 L15,8 L15,4 L7,4 L7,8 L5,8 L5,4 C5,2.94686456 5.8170484,2.08194636 6.85089728,2.00549334 L7,2 L16.0002136,2 Z M3,11 L3,16 C3,17.105 4.03269231,18 5.30769231,18 L15.6923077,18 C16.9673077,18 18,17.105 18,16 L18,11 C18,9.895 16.9673077,9 15.6923077,9 L5.30769231,9 C4.03269231,9 3,9.895 3,11 Z M14.5,11 C15.328,11 16,11.672 16,12.5 C16,13.328 15.328,14 14.5,14 L14,14 L14,16 L13,16 L13,11 L14.5,11 Z M12,11 L12,12 L11,12 L11,16 L10,16 L10,12 L9,12 L9,11 L12,11 Z M8,11 L8,12 L6,12 L6,13 L7.6,13 L7.6,14 L6,14 L6,16 L5,16 L5,11 L8,11 Z M14.5,12 L14,12 L14,13 L14.5,13 C14.776,13 15,12.776 15,12.5 C15,12.224 14.776,12 14.5,12 Z"></path>
            </g>
        </g>
    </g>
</svg>
);

export const AYIcon = withFdsIconWrapper(AY);

<Preview>
  <Story name="Caret">
    <StoryWrapper>
      <StoryItem>
        <Button theme="blue" hasCaret label="Caret" />
      </StoryItem>
      <StoryItem>
        <IconButton Icon={AYIcon} />
      </StoryItem>
    </StoryWrapper>
  </Story>
</Preview>

### Loading

Loading states should be shown whenever an action has the ability 
to take more than a couple of seconds of loading time

<Preview>
  <Story name="Loading">
    <>
      <StoryWrapper>
      {THEMES.map(theme => (
        <StoryItem>
          <Button
            theme={theme}
            isLoading
            label="Button"
          />
        </StoryItem>
      ))}
      {THEMES.map(theme => (
        <StoryItem>
          <Button
            theme={theme}
            isDestructive
            isLoading
            label="Button"
          />
        </StoryItem>
      ))}
      </StoryWrapper>
      <div class="inverted bgColor--blue padding--all rounded--all margin--top">
        <StoryWrapper>
          <StoryItem>
            <Button theme="ghost" isLoading label="Button" />
          </StoryItem>
          <StoryItem>
            <Button theme="outlined" isLoading label="Button" />
          </StoryItem>
        </StoryWrapper>
      </div>
    </>
  </Story>
</Preview>

### Misc

<Preview>
  <Story name="Misc">
    <StoryWrapper>
      <StoryItem>
        <div style={{ width: '100px' }}>
          <Button label="Text can wrap" />
        </div>
      </StoryItem>
    </StoryWrapper>
  </Story>
</Preview>
