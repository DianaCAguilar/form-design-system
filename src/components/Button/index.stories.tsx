import React from 'react';
import State from 'util/storybook-docs/State';
import { StoryWrapper, StoryItem } from 'util/storybook-docs/StoryLayout';
import { StarFilledIcon, WorldIcon } from 'lib/icons/react';
import Button, { ButtonProps, THEMES, SIZES } from '.';

const Template = (args): JSX.Element => <Button label="Button" {...args} />;

export const Primary = Template.bind({});

export const Themes = (): JSX.Element => (
  <StoryWrapper>
    {THEMES.map((theme, idx) => (
      <StoryItem key={idx}>
        <Button theme={theme} label={theme} />
      </StoryItem>
    ))}
  </StoryWrapper>
);

export const isDestructive = (): JSX.Element => (
  <StoryWrapper>
    {THEMES.map((theme, idx) => (
      <StoryItem key={idx}>
        <Button isDestructive theme={theme} label={theme} />
      </StoryItem>
    ))}
  </StoryWrapper>
);

isDestructive.parameters = {
  docs: {
    description: {
      story:
        'Modifies the theme. Should only be used when the interaction involves data loss, and should also prompt the user to confirm the action.',
    },
  },
};

export const Inverted = (): JSX.Element => (
  <div className="inverted bgColor--blue padding--all rounded--all">
    <StoryWrapper>
      <StoryItem>
        <Button theme="ghost" label="Button" />
      </StoryItem>
      <StoryItem>
        <Button theme="outlined" label="Button" />
      </StoryItem>
    </StoryWrapper>
  </div>
);

Inverted.parameters = {
  docs: {
    description: {
      story:
        'We currently support inverting `ghost` and `outlined` theme buttons. Simply surround the button(s) with an `.inverted` class.',
    },
  },
};

export const isActive = (): JSX.Element => (
  <>
    <StoryWrapper>
      <State
        initialValue={{}}
        render={(value, setValue) => (
          <>
            {THEMES.map((theme, idx) => (
              <StoryItem key={idx}>
                <Button
                  theme={theme}
                  isActive={!value[idx]}
                  label={theme}
                  onClick={() => setValue({ ...value, [idx]: !value[idx] })}
                />
              </StoryItem>
            ))}
          </>
        )}
      />
      <State
        initialValue={{}}
        render={(value, setValue) => (
          <>
            {THEMES.map((theme, idx) => (
              <StoryItem key={idx}>
                <Button
                  theme={theme}
                  isActive={!value[idx]}
                  isDestructive
                  label={theme}
                  onClick={() => setValue({ ...value, [idx]: !value[idx] })}
                />
              </StoryItem>
            ))}
          </>
        )}
      />
    </StoryWrapper>
    <div className="inverted bgColor--blue padding--all rounded--all margin--top">
      <StoryWrapper>
        <State
          initialValue={{}}
          render={(value, setValue) => (
            <>
              {['ghost', 'outlined'].map((theme, idx) => (
                <StoryItem key={idx}>
                  <Button
                    theme={theme}
                    isActive={!value[idx]}
                    label={theme}
                    onClick={() => setValue({ ...value, [idx]: !value[idx] })}
                    data-test="hey"
                  />
                </StoryItem>
              ))}
            </>
          )}
        />
      </StoryWrapper>
    </div>
  </>
);

isActive.parameters = {
  docs: {
    description: {
      story:
        '`isActive` should be used on buttons to signify the button having some affect (applying a filter, causing a popover to be open). **Click to toggle on and off**.',
    },
  },
};

export const Sizes = (): JSX.Element => (
  <>
    {SIZES.map((size, idx) => (
      <div key={idx} className="display--inlineBlock margin--right--s margin--bottom--s">
        <Button size={size} label="Button" />
      </div>
    ))}
  </>
);

export const FullWidth = Template.bind({});
FullWidth.args = {
  isFullWidth: true,
};

export const Icons = (): JSX.Element => (
  <StoryWrapper>
    <StoryItem>
      <Button IconLeft={WorldIcon} label="Button" />
    </StoryItem>
    <StoryItem>
      <Button IconRight={StarFilledIcon} label="Button" />
    </StoryItem>
  </StoryWrapper>
);

Icons.parameters = {
  docs: {
    description: {
      story:
        'ONLY pass in FDS Icon references to `IconLeft` or `IconRight`. Very important. (refer to code sample)',
    },
  },
};

export const Caret = (): JSX.Element => (
  <StoryWrapper>
    <StoryItem>
      <Button theme="blue" hasCaret label="Caret" />
    </StoryItem>
    <StoryItem>
      <Button theme="blue" hasCaret IconRight={StarFilledIcon} label="Caret" />
    </StoryItem>
    <StoryItem>
      <Button theme="blue" hasCaret IconRight={StarFilledIcon} label="Caret" />
    </StoryItem>
  </StoryWrapper>
);

export const Loading = (): JSX.Element => (
  <>
    <StoryWrapper>
      {THEMES.map((theme, idx) => (
        <StoryItem key={idx}>
          <Button theme={theme} isLoading label="Button" />
        </StoryItem>
      ))}
      {THEMES.map((theme, idx) => (
        <StoryItem key={idx}>
          <Button theme={theme} isDestructive isLoading label="Button" />
        </StoryItem>
      ))}
    </StoryWrapper>
    <div className="inverted bgColor--blue padding--all rounded--all margin--top">
      <StoryWrapper>
        <StoryItem>
          <Button theme="ghost" isLoading label="Button" />
        </StoryItem>
        <StoryItem>
          <Button theme="outlined" isLoading label="Button" />
        </StoryItem>
      </StoryWrapper>
    </div>
  </>
);

Loading.parameters = {
  docs: {
    description: {
      story:
        'Loading states should be shown whenever an action has the ability to take more than a couple of seconds of loading time',
    },
  },
};

export const Breakout = (args: ButtonProps): JSX.Element => (
  <StoryWrapper>
    <StoryItem>
      <div
        className="alignChild--center--center padding--all--xl"
        style={{ border: '1px dashed #f09', position: 'relative' }}
      >
        <Button {...args} label="Button" isBreakoutLink />
      </div>
    </StoryItem>
  </StoryWrapper>
);

Breakout.parameters = {
  docs: {
    description: {
      story: 'Breakout buttons extend their clickable area to the next nearest parent.',
    },
  },
};

export const Misc = (): JSX.Element => (
  <div style={{ width: '100px' }}>
    <Button label="Text can wrap" />
  </div>
);

export default {
  component: Button,
  title: 'components/Button',
  parameters: {
    docs: {
      fds: {
        attributes: ['rest'],
      },
    },
  },
};
