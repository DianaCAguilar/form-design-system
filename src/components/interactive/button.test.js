import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

describe('ButtonGroup component', () => {

  it('matches snapshot (default props)', () => {
    const component = shallow(<Button label="Button" />);
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot (set all props)', () => {
    const component = shallow(<Button 
      label="Button" Link={() => {}} isLoading isDestructive disabled IconLeft={() => {}} IconRight={() => {}} isFullWidth theme='outlined' hasCaret
    />)
    expect(component).toMatchSnapshot();
  });
});
