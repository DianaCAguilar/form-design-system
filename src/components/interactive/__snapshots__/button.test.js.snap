// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup component matches snapshot (default props) 1`] = `
<button
  className="fdsButton fontStyle--caps display--inlineFlex rounded--all alignChild--center--center fdsButton--m fdsButton--blue"
>
  <span
    className=""
  >
    Button
  </span>
</button>
`;

exports[`ButtonGroup component matches snapshot (set all props) 1`] = `
<Link
  className="fdsButton fontStyle--caps display--inlineFlex rounded--all alignChild--center--center fdsButton--m fdsButton--outlined fdsButton--isDestructive fdsButton--loading fdsButton--disabled fdsButton--isFullWidth"
  disabled={false}
>
  <span
    className="fdsButton--hidden"
  />
  <div
    className="alignChild--center--center fdsButton-icon--left margin--right--half fdsButton--hidden"
  >
    <Icon
      customSize={16}
    />
  </div>
  <div
    className="margin--left--half alignChild--center--center"
  >
    <CaretDownIcon
      customSize={12}
    />
  </div>
</Link>
`;
